entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityOperationService.kt

```diff
@@ -58,6 +57,87 @@ class EntityOperationService(
         return BatchOperationResult(ArrayList(success), ArrayList(errors))
     }
 
+    /**
+     * Update a batch of [entities].
+     * Only entities with relations linked to existing entities will be updated.
+     *
+     * @return a [BatchOperationResult] with list of updated ids and list of errors (either not totally updated or
+     * linked to invalid entity).
+     */
+    fun update(entities: List<ExpandedEntity>, createBatchResult: BatchOperationResult): BatchOperationResult {
+        val existingEntitiesIds = createBatchResult.success.plus(entities.map { it.id })
+        val nonExistingEntitiesIds = createBatchResult.errors.map { it.entityId }
+        return entities.parallelStream().map { entity ->
+            updateEntity(entity, existingEntitiesIds, nonExistingEntitiesIds)
+        }.collect(
+            { BatchOperationResult() },
+            { batchOperationResult, updateResult ->
+                updateResult.fold({
+                    batchOperationResult.errors.add(it)
+                }, {
+                    batchOperationResult.success.add(it)
+                })
+            },
+            BatchOperationResult::plusAssign
+        )
+    }
+
+    private fun updateEntity(
+        entity: ExpandedEntity,
+        existingEntitiesIds: List<String>,
+        nonExistingEntitiesIds: List<String>
+    ): Either<BatchEntityError, String> {
+        // All new attributes linked entities should be existing in the DB.
+        val linkedEntitiesIds = entity.getLinkedEntitiesIds()
+        val invalidLinkedEntityId =
+            findInvalidEntityId(linkedEntitiesIds, existingEntitiesIds, nonExistingEntitiesIds)
+
+        // If there's a link to an invalid entity, then avoid calling the processor and return an error
+        if (invalidLinkedEntityId != null) {
+            return Either.left(
+                BatchEntityError(
+                    entity.id,
+                    arrayListOf("Target entity $invalidLinkedEntityId does not exist.")
+                )
+            )
+        }
+
+        return try {
+            val (_, notUpdated) = entityService.appendEntityAttributes(
+                entity.id,
+                entity.attributes,
+                false
+            )
+
+            if (notUpdated.isEmpty()) {
+                Either.right(entity.id)
+            } else {
+                Either.left(
+                    BatchEntityError(
+                        entity.id,
+                        ArrayList(notUpdated.map { it.attributeName + " : " + it.reason })
+                    )
+                )
+            }
+        } catch (e: BadRequestDataException) {
+            Either.left(BatchEntityError(entity.id, arrayListOf(e.message)))
+        }
+    }
+
+    private fun findInvalidEntityId(
```

Comment:
a `hasInvalidLinkedEntity(...): Boolean` would make it clearer (because, at the end, you are not interested by the returned string)

Reply:
I did it this way to track the invalid entity and to find the issue faster. This can be changed if we decide it's useless.

Reply:
If you use the returned (invalid) entity id (for instance, for an error message in the returned batch result), that's OK for me.

If you don't, switch to a boolean.